util/netevent.c

```diff
@@ -894,8 +910,22 @@ comm_point_udp_ancil_callback(int fd, short event, void* arg)
 					sizeof(struct in_addr));
 				break;
 #endif /* IP_PKTINFO or IP_RECVDSTADDR */
+#ifdef HAVE_LINUX_NET_TSTAMP_H
+			} else if( cmsg->cmsg_level == SOL_SOCKET &&
+				cmsg->cmsg_type == SO_TIMESTAMPNS) {
+				ts = (struct timespec *)CMSG_DATA(cmsg);
+				TIMESPEC_TO_TIMEVAL(&rep.c->recv_tv, ts);
+			} else if( cmsg->cmsg_level == SOL_SOCKET &&
+				cmsg->cmsg_type == SO_TIMESTAMPING) {
+				ts = (struct timespec *)CMSG_DATA(cmsg);
+				TIMESPEC_TO_TIMEVAL(&rep.c->recv_tv, ts);
+			} else if( cmsg->cmsg_level == SOL_SOCKET &&
+				cmsg->cmsg_type == SO_TIMESTAMP) {
+				rep.c->recv_tv = *(struct timeval *)CMSG_DATA(cmsg);
```

Comment:
Even though I do not know of platforms with both timestamping and alignment issues, the code change is to be careful with unaligned memory reads.
```suggestion
				memmove(&rep.c->recv_tv, (struct timeval *)CMSG_DATA(cmsg), sizeof(struct timeval));
```

Reply:
Even though I'm more about to rely on compiler in such cases, I'm ok with this change. But there is no need for cast anymore with memmove.

Reply:
Yes, that is true. I'll mark this as resolved then.